Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1

```diff
@@ -380,6 +500,332 @@ function Wait-ServiceFabricApplicationTypeStatusChange
         -RetryMessage (Get-VstsLocString -Key SFSDK_RetryingGetApplicationType)
 }
 
+function Wait-ServiceFabricApplicationTypeRegistrationStatus
+{
+    Param (
+        [string]
+        $ApplicationTypeName,
+
+        [string]
+        $ApplicationTypeVersion,
+
+        [int]
+        $TimeoutSec
+    )
+
+    $global:operationId = $SF_Operations.GetApplicationType
+    $getAppTypeAction = { Get-ServiceFabricApplicationTypeAction -ApplicationTypeName $ApplicationTypeName -ApplicationTypeVersion $ApplicationTypeVersion }
+    $getAppTypeRetryEvaluator = {
+        param($appType)
+
+        # If provisioning not started, retry register
+        if(!$appType)
+        {
+            return $true
+        }
+        # if app type is provisioned, don't retry
+        elseif($appType.Status -eq [System.Fabric.Query.ApplicationTypeStatus]::Available)
+        {
+            return $false
+        }
+        # if app type exist and if its status has not changed to a terminal one, do retry
+        elseif(($appType.Status -eq [System.Fabric.Query.ApplicationTypeStatus]::Provisioning) -or ($appType.Status -eq [System.Fabric.Query.ApplicationTypeStatus]::Unprovisioning))
+        {
+            return $true
+        }
+        # if app type exist and if its status has changed to a terminal one, throw
```

Comment:
I had this minor comment in previous PR: why is the elseif condition required. It should always throw if reached here

Reply:
Yes, it will always throw